### PR TITLE
Implement the "announcements" attributes from Grouper

### PR DESCRIPTION
**Pull request by michho8**
# Ticket Link

[Ticket 1532](https://uhawaii.atlassian.net/browse/GROUPINGS-1532)

# List of squashed commits

- Two containers but logo offset
- Create GroupingsAnnouncement.java and GroupingsAnnouncements.java
- Saving progress
- Commit to pull from main
- Logic correct
- Created asObjectList method for JsonUtil
- Reorganize Announcements
- Revert "Created asObjectList method for JsonUtil"
- This reverts commit https://github.com/uhawaii-system-its-ti-iam/uh-groupingsapi/commit/1257ec8fdad39e2cac5525743dcdd010537528e6.
- Get announcement object to show as null on site
- All announcements show on browser
- Implement date comparisons, need new test data
- Get rid of user param
- Create helper methods in Announcement
- Delete PlannedOutage files
- Clean up code
- Start integration testing
- Whole response body display on site (resultCode and announcements)
- Display one valid message and clean up code
- No other errors other than currentUser
- Clean up code
- Convert to showing all Active messages
- Clean up code and rename functions
- Add integration test and encapsulate Grouper attr paths
- Clean up imports
- Unit test for the Announcements obj
- Unit test for Announcement
- Clean up code and fix tests
- Merge conflicts again
- Address pr suggestions
- Get rid of Codacy unused imports
- Undo renaming files
- Address Github feedback
- Get rid of redundant findAnnouncements call in Announcements.java
- Get rid of findAnnouncements altogether for Codacy
- Change setAnnouncements() to use getResult instead of getResults
- Address Github feedback
- Get rid of EmailService changes
- Fix import order
- Codacy unused variable
- Imports
- Update email service files to match project's main
- Rebranched
- Fix comments and imports
- Fix import order
- Update AnnouncementsTest with new data

# Test Checklist

- [x] Exhibits Clear Code Structure:
- [x] Project Unit Tests Passed:
- [x] Project Integration Tests Passed:
- [x] Executes Expected Functionality:
- [x] Tested For Incorrect/Unexpected Inputs:

### CODE REVIEW

File: src/main/java/edu/hawaii/its/api/controller/GroupingsRestControllerv2_1.java [hunk 1]
```diff
@@ -4,6 +4,7 @@
 
 import javax.annotation.PostConstruct;
 
+import edu.hawaii.its.api.service.*;
```
Comment (JorWo) [inline src/main/java/edu/hawaii/its/api/controller/GroupingsRestControllerv2_1.java:7]: Remove * import, thanks!

File: src/main/java/edu/hawaii/its/api/type/Announcements.java [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+        return announcements;
+    }
+    public String getResultCode() {
+        return resultCode;
+    }
+
+    public List<String> validMessages(List<Announcement> allGroupingsAnnouncements) {
+        List<String> validMessages = new ArrayList<>();
+        for (Announcement groupingsAnnouncement : allGroupingsAnnouncements) {
+            //check whether each announcement message is of an Active state
```
Comment (JorWo) [inline src/main/java/edu/hawaii/its/api/type/Announcements.java:39]: I think this comment is not needed, the code below is comment is self-explanatory.

File: src/main/java/edu/hawaii/its/api/service/AnnouncementsService.java [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+    @Value("${groupings.api.propertystring}")
+    private String ANNOUNCEMENTS_ATTR_DEF;
+
+    @Autowired
+    private GrouperApiService grouperApiService;
+
+    public Announcements setAnnouncements(AttributesResult attributesResult) {
+        //hydrate announcements by creating a list of Announcement.class objects
+        List<Announcement> announcementsList = new ArrayList<>();
+        announcementsList = JsonUtil.asList(attributesResult.getDescription(), Announcement.class); //seems to be initializing announcement
```
Comment (JorWo) [inline src/main/java/edu/hawaii/its/api/service/AnnouncementsService.java:32]: These two comments are not needed either, the code under is it self-explanatory.

**Comment by JorWo** [comment]
For the rest of the comments, could you put it in this format? `// This is a comment.` Notice the space after the `//` and formal sentence structure. Thanks!

File: src/test/java/edu/hawaii/its/api/type/AnnouncementsTest.java [hunk 4]
```diff
@@ -0,0 +1,64 @@
+package edu.hawaii.its.api.type;
+
+import java.time.LocalDateTime;
+import java.util.ArrayList;
+import java.util.List;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import static org.junit.jupiter.api.Assertions.*;
```
Comment (JorWo) [inline src/test/java/edu/hawaii/its/api/type/AnnouncementsTest.java:8]: Remove the * import here, thanks!

**Comment by JorWo** [comment]
Be sure to remove all * imports, thanks!

**Review by mhodgesatuh — APPROVED** [review]
I didn't see any issues.

**Comment by JorWo** [comment]
A few last changes:

- [ ]  Add a newline in between the java, javax, com, org, edu imports. Static imports taking priority at the top. You can also import the new code style file that Isaac updated and it should fix up the imports for you. 
- [ ] Remove the * imports, there are still some * imports in the test classes.

Thanks!

**Review by JorWo — APPROVED** [review]
Great job! I will wait on your UI PR before merging this.

**Pull request merged**